diff --git a/BackwardConfig.cmake b/BackwardConfig.cmake
index 6cff86c..bb24c46 100644
--- a/BackwardConfig.cmake
+++ b/BackwardConfig.cmake
@@ -111,7 +111,7 @@ if (${STACK_DETAILS_AUTO_DETECT})
 		set(STACK_DETAILS_BACKTRACE_SYMBOL FALSE)
 	elseif(LIBDWARF_FOUND)
 		LIST(APPEND _BACKWARD_INCLUDE_DIRS ${LIBDWARF_INCLUDE_DIRS})
-		LIST(APPEND BACKWARD_LIBRARIES ${LIBDWARF_LIBRARIES})
+		LIST(APPEND _BACKWARD_LIBRARIES ${LIBDWARF_LIBRARIES})
 
 		set(STACK_DETAILS_DW FALSE)
 		set(STACK_DETAILS_BFD FALSE)
@@ -133,7 +133,7 @@ else()
 	endif()
 
 	if (STACK_DETAILS_DWARF)
-		LIST(APPEND _BACKWARD_LIBRARIES dwarf elf)
+		LIST(APPEND _BACKWARD_LIBRARIES CONAN_PKG::libdwarf ${CMAKE_DL_LIBS})
 	endif()
 endif()
 
@@ -196,7 +196,7 @@ if (NOT TARGET Backward::Backward)
 	)
 	if(BACKWARD_HAS_EXTERNAL_LIBRARIES)
 		set_target_properties(Backward::Backward PROPERTIES
-			INTERFACE_LINK_LIBRARIES "${BACKWARD_LIBRARIES}" 
+			INTERFACE_LINK_LIBRARIES "${BACKWARD_LIBRARIES}"
 		)
 	endif()
 endif()
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 119d390..f33977e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,9 +20,21 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.0)
 project(backward CXX)
 
+# Introduce variables:
+# * CMAKE_INSTALL_LIBDIR
+# * CMAKE_INSTALL_BINDIR
+# * CMAKE_INSTALL_INCLUDEDIR
+include(GNUInstallDirs)
+
+if (EXISTS ${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+    include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+    conan_basic_setup(TARGETS)
+	set(USING_CONAN TRUE)
+endif()
+
 include(BackwardConfig.cmake)
 
 # check if compiler is nvcc or nvcc_wrapper
@@ -74,6 +86,7 @@ endif()
 add_library(backward ${libtype} backward.cpp)
 target_compile_definitions(backward PUBLIC ${BACKWARD_DEFINITIONS})
 target_include_directories(backward PUBLIC ${BACKWARD_INCLUDE_DIRS})
+conan_target_link_libraries(backward)
 
 ###############################################################################
 # TESTS
@@ -119,9 +132,9 @@ endif()
 
 install(
     FILES "backward.hpp"
-    DESTINATION ${CMAKE_INSTALL_PREFIX}/include
+    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
 )
 install(
     FILES "BackwardConfig.cmake"
-    DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/backward
-)
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}/backward
+)
\ No newline at end of file
diff --git a/backward.hpp b/backward.hpp
index c9a922f..29b4bf9 100644
--- a/backward.hpp
+++ b/backward.hpp
@@ -1922,7 +1922,7 @@ private:
 
 		dwarf_file_t file_handle;
 		file_handle.reset(open(filename_object.c_str(), O_RDONLY));
-		if (file_handle < 0) {
+		if (file_handle == nullptr) {
 			return r;
 		}
 
